ci/base-tests.sh

```diff
@@ -29,26 +31,27 @@ export CARGO_TARGET_DIR=`pwd`/target/
 (
   # Check sysroot handling
   sysroot=$(./target/debug/clippy-driver --print sysroot)
-  test $sysroot = $(rustc --print sysroot)
+  test "$sysroot" = "$(rustc --print sysroot)"
 
-  if [ -z $OS_WINDOWS ]; then
+  if [[ -z "$OS_WINDOWS" ]]; then
     desired_sysroot=/tmp
   else
     desired_sysroot=C:/tmp
   fi
   sysroot=$(./target/debug/clippy-driver --sysroot $desired_sysroot --print sysroot)
-  test $sysroot = $desired_sysroot
+  test "$sysroot" = $desired_sysroot
 
   sysroot=$(SYSROOT=$desired_sysroot ./target/debug/clippy-driver --print sysroot)
-  test $sysroot = $desired_sysroot
+  test "$sysroot" = $desired_sysroot
 
   # Make sure this isn't set - clippy-driver should cope without it
   unset CARGO_MANIFEST_DIR
 
   # Run a lint and make sure it produces the expected output. It's also expected to exit with code 1
-  # XXX How to match the clippy invocation in compile-test.rs?
-  ! ./target/debug/clippy-driver -Dwarnings -Aunused -Zui-testing --emit metadata --crate-type bin tests/ui/cstring.rs 2> cstring.stderr
```

Comment:
I don't understand what this command does. Could someone explain it to me?

Reply:
This whole block checks if `clippy-driver` can be run without cargo. 

This command is just a test, if the output of `clippy-driver` is the same as the output of the ui-test. Why `cstring.rs` was chosen for this test, I have no idea. 

I guess all these flags are there to simulate the behavior of `compiletest-rs`.

Reply:
The PR that added this has some more context. Maybe include the PR link in the comment?